fhir-audit/src/main/java/com/ibm/fhir/audit/configuration/handlers/ComponentIpHandler.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.audit.configuration.handlers;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.config.FHIRConfigHelper;
+import com.ibm.fhir.config.FHIRConfiguration;
+
+/**
+ * Calculates the Component Ip and subsequent retrieval of the IPs
+ */
+public class ComponentIpHandler {
+    private static final String CLASSNAME = ComponentIpHandler.class.getName();
+    private static final Logger log = java.util.logging.Logger.getLogger(CLASSNAME);
+
+    private static final String IP = determineIP();
+
+    public ComponentIpHandler() {
+        // No Operation
+    }
+
+    /*
+     * determines the IP
+     */
+    private static String determineIP() {
```

Comment:
Please see FHIRException.getUniqueId() and determine if there is common functionality that we can use. At least we should be using the same approach in the two places I think.

Reply:
The problem with getUniqueId is it uses a lookup to get the LocalHost and can hang, specifically on lookup of the main hostname. This is a bit more resilient and faster. It also returns all IP addresses, where as the uniqueId only returns one.

Reply:
I've moved the handler to core